docs/Getting Started/Debian/Debian Bullseye Root on ZFS.rst

```diff
@@ -951,10 +956,139 @@ Step 6: First Boot
    Replace ``YOUR_USERNAME`` with your desired username::
 
      username=YOUR_USERNAME
+     
+   Choose one of the following:
+     
+   - Unencrypted homedir or whole-disk encryption
+     
+       ::
+
+         zfs create rpool/home/$username
+         adduser $username
+       
+   - Encrypted homdir per user, automatic decryption on login
+     
+       ::
+       
+         zfs create rpool/home/$username -o encryption=on -o keyformat=passphrase -o keylocation=prompt -o canmount=noauto -o org.openzfs.systemd:ignore=on
+         zfs mount rpool/home/$username
+         adduser $username
+         
+       **Note**: Use the same strong password for ZFS encryption and user password. Please note: After a breach of your password changing the ZFS password does not restore protection.
+       
+       Tell PAM to unlock the dataset key on login::
+       
+         printf "auth [success=1 default=ignore] pam_succeed_if.so service = sudo quiet\n\
+                 auth optional pam_exec.so expose_authtok /usr/local/sbin/unlock-key-zfs-homedir\n"\
+           >> /etc/pam.d/common-auth
+           
+       The second line calls a new unlock-script, the first line disables it if we just sudo.
+       
+       Create the unlock-script::
+       
+         touch /usr/local/sbin/unlock-key-zfs-homedir 
+         chmod a+x /usr/local/sbin/unlock-key-zfs-homedir 
+         vi /usr/local/sbin/unlock-key-zfs-homedir 
+       
+       and put into it::
```

Comment:
1. shellcheck on this
2. I don't see a reason to require bash over plain sh.
3. Is there a particular reason we couldn't allow for an encrypted root home directory?

I get this instead:
```
#!/bin/sh

set -e

if [ "$PAM_USER" = "root" ]
then
    dataset=rpool/root
    mountpoint=/root
else
    dataset=rpool/home/$PAM_USER
    mountpoint=/home/$PAM_USER
fi

# Exit if already mounted.
findmnt "$mountpoint" && exit 0

# If the dataset does not exist, this will exit.
encryption=$(zfs list "$dataset" -o encryption -H 2> /dev/null) || exit 0

[ "$encryption" = off ] && exit 0

zfs load-key "rpool/home/$PAM_USER" < /dev/stdin
```

I'm not sure if that `< /dev/stdin` is accomplishing anything. Wouldn't stdin be connected to `zfs load-key` by default?

Reply:
I think it's not wise to allow for an encrypted root home directory, because (see above) systemd doesn't start a user session when using sudo, so root would end up without mounted home directory.

Reply:
BTW: I was pretty surprised `zfs load-key` worked without explicit `< /dev/stdin`.